src/main/java/org/apache/jackrabbit/filevault/maven/packaging/GenerateMetadataMojo.java

```diff
@@ -868,7 +884,53 @@ private Properties computeProperties(String dependenciesString) {
         return fileMap;
     }
 
+    /**
+     * Establishes a session-shareable workDirectory lookup map for the given pluginContext.
+     *
+     * @param pluginContext a Map retrieved from {@link MavenSession#getPluginContext(PluginDescriptor, MavenProject)}.
```

Comment:
I use MavenProject.getProperties().put(...) for other stuff which is shared between different goals of the same plugin but it seems that PluginContext is actually more appropriate here (compare also with https://issues.apache.org/jira/browse/JCRVLT-326)

Reply:
Yes, it looks like getPluginContext is much easier for same-plugin communication between project modules, whereas the getProperties() solution works better for same-module communication between plugins.